tests/test_reporters.py

```diff
@@ -55,3 +55,41 @@ def test_vtk_reporter(tmpdir):
     simulation.step(2)
     assert os.path.isfile(tmpdir/"output_00000000.vtr")
     assert os.path.isfile(tmpdir/"output_00000001.vtr")
+
+
+@pytest.mark.parametrize("Flow", [TaylorGreenVortex2D,TaylorGreenVortex3D,DecayingTurbulence2D])
+def test_EnergySpectrumReporter(tmpdir, Flow):
+    lattice = Lattice(D2Q9, device='cpu')
+    if Flow == TaylorGreenVortex3D:
+        lattice = Lattice(D3Q27, device='cpu')
+    flow = Flow(resolution=20, reynolds_number=1600, mach_number=0.01, lattice=lattice)
+    collision = BGKCollision(lattice, tau=flow.units.relaxation_parameter_lu)
+    streaming = StandardStreaming(lattice)
+    simulation = Simulation(flow=flow, lattice=lattice, collision=collision, streaming=streaming)
+    simulation.reporters.append(SpectrumReporter(lattice, flow, out=None))
+    simulation.reporters.append(EnergyReporter(lattice, flow, out=None))
+    simulation.step(1)
+    spectrum = simulation.reporters[0].out
+    energy = simulation.reporters[1].out
+
+    if Flow == DecayingTurbulence2D:
+        dx = flow.units.convert_length_to_pu(1.0)
+        kx = np.fft.fftfreq(flow.resolution, d=1 / flow.resolution)
+        ky = np.fft.fftfreq(flow.resolution, d=1 / flow.resolution)
+        kx, ky = np.meshgrid(kx, ky)
+        kk = np.sqrt(kx ** 2 + ky ** 2)
+        kk[0][0] = 1e-16
+        ek = (kk) ** 4 * np.exp(-2 * (kk / flow.k0) ** 2)
+        ek[0][0] = 0
+        ek /= np.sum(ek)
+        ek *= flow.ic_energy
+        ek_ref = np.zeros(int(np.max(kk)))
+        k = np.zeros(int(np.max(kk)))
+        for wv in range(int(np.max(kk))):
+            ii, jj = np.where((kk > (wv - 0.5)) & (kk < (wv + 0.5)))
+            ek_ref[wv] = np.sum(ek[ii, jj])
```

Comment:
Is this copied from the initialization of DecayingTurbulence? Why not make the initial spectrum accessible as a property? This would make it clearer what happens here.

Reply:
done in separate PR